### PR TITLE
docs: fix vue-tsc version in TypeScript page

### PR DESCRIPTION
**Pull request by rileychh**
### 🔗 Linked issue

None

### 📚 Description

[The TypeScript page in Key Concepts](https://nuxt.com/docs/guide/concepts/typescript) currently recommends an outdated version of `vue-tsc`, causing type-checking issues with Nuxt 3.12.4.

```
$ pnpm add -D vue-tsc@^1 typescript

# ...

devDependencies:
+ typescript 5.5.4
+ vue-tsc 1.8.27 (2.0.29 is available)

 WARN  Issues with peer dependencies found
.
└─┬ nuxt 3.12.4
  └─┬ @nuxt/vite-builder 3.12.4
    └─┬ vite-plugin-checker 0.7.2
      └── ✕ unmet peer vue-tsc@>=2.0.0: found 1.8.27

$ pnpm exec nuxi typecheck

<project-path>/node_modules/.pnpm/vue-tsc@1.8.27_typescript@5.5.4/node_modules/vue-tsc/bin/vue-tsc.js:68
			throw err;
			^
Search string not found: "for (const existingRoot of buildInfoVersionMap.roots) {"
(Use `node --trace-uncaught ...` to show where the exception was thrown)

Node.js v20.16.0
```

Installing `vue-tsc@^2` or `vue-tsc@latest` resolves the issue. I have updated the documentation to guide users to install the correct version of `vue-tsc`.

### CODE REVIEW

**Review by danielroe — APPROVED** [review]

File: docs/2.guide/1.concepts/8.typescript.md [hunk 1]
```diff
@@ -12,19 +12,19 @@ To enable type-checking at build or development time, install `vue-tsc` and `typ
 ::code-group
 
   ```bash [yarn]
-  yarn add --dev vue-tsc@^1 typescript
+  yarn add --dev vue-tsc@^2 typescript
   ```
 
   ```bash [npm]
-  npm install --save-dev vue-tsc@^1 typescript
+  npm install --save-dev vue-tsc@^2 typescript
```
Comment (danielroe) [inline docs/2.guide/1.concepts/8.typescript.md:19]: We can probably remove the major version now.

Comment (rileychh) [reply docs/2.guide/1.concepts/8.typescript.md]: Sure, I removed the version in [a20a3bb](https://github.com/nuxt/nuxt/pull/28484/commits/a20a3bb64f335c1367d4f735e9e9c5a3a758fabb)!

**Comment by vbrajon** [comment]
Thanks for this doc fix, I wanted to do it but saw it already fixed.
It seems not to be propagated still (3 days ago), is it expected ?

**Comment by danielroe** [comment]
It will be when I merge it into the 3.x branch.

**Pull request merged**